test/e2e/framework/util.go

```diff
@@ -2757,26 +2761,87 @@ func WaitForPodsWithLabel(c clientset.Interface, ns string, label labels.Selecto
 	return
 }
 
-// DeleteRCAndPods a Replication Controller and all pods it spawned
-func DeleteRCAndPods(clientset clientset.Interface, internalClientset internalclientset.Interface, ns, name string) error {
-	By(fmt.Sprintf("deleting replication controller %s in namespace %s", name, ns))
-	rc, err := clientset.Core().ReplicationControllers(ns).Get(name)
+func getSelectorAndReplicasFromResource(c clientset.Interface, resource, ns, name string) (*unversioned.LabelSelector, int32, error) {
+	var rc *v1.ReplicationController
+	var dep *extensions.Deployment
+	var rs *extensions.ReplicaSet
+	var selector *unversioned.LabelSelector
+	var replicas int32
+	var err error
+
+	switch resource {
+	case ReplicationControllerResource:
+		rc, err = c.Core().ReplicationControllers(ns).Get(name)
+		if err == nil {
+			selector = &unversioned.LabelSelector{MatchLabels: rc.Spec.Selector}
+			if rc.Spec.Replicas != nil {
+				replicas = *rc.Spec.Replicas
+			}
+		}
+	case ReplicaSetResource:
+		rs, err = c.Extensions().ReplicaSets(ns).Get(name)
+		if err == nil {
+			selector = rs.Spec.Selector
+			if rc.Spec.Replicas != nil {
+				replicas = *rs.Spec.Replicas
+			}
+		}
+	case DeploymentResource:
+		dep, err = c.Extensions().Deployments(ns).Get(name)
+		if err == nil {
+			selector = dep.Spec.Selector
+			if rc.Spec.Replicas != nil {
+				replicas = *dep.Spec.Replicas
+			}
+		}
+	}
+	return selector, replicas, err
```

Comment:
Hmm - I would prefer a different pattern. So I think it would be much more clear to have this method just return:
```
"(runtime.Object, error)"
```
(where this runtime.Object would be either Deployment, RS or RC).

And then we would have two separate methods:
```
getSelectorFromObject(runtime.Object) unversioned.Label.Selector
getReplicas(runtime.Object) int
```
[or whatever you call them].

Then I think it's much cleaner.

Also in those methods "getSelector" you no longer depend on kind name - it would be just: sth like:
```
switch typed := obj.(type):
case *v1.ReplicationController:
  return typed.Spec.Selector
...
}
```